src/lib/components/edit-transaction.svelte

```diff
@@ -79,41 +79,43 @@
 		if (!authStore.user) {
 			return
 		}
-		await adapter.addTransaction({
-			user_id: authStore.user?.id,
-			investment_id: investment.id,
-			type: transaction.type,
-			amount,
-			label,
-			date: date.toDateString(),
-			repeat: isRecurring ? repeat : null,
-			repeat_unit: isRecurring ? repeatUnit : null,
-			end_date: isRecurring ? endDate.toDateString() : null,
-		})
+		if (investment)
+			await adapter.addTransaction({
+				user_id: authStore.user?.id,
+				investment_id: investment.id,
+				type: transactionType,
+				amount,
+				label,
+				date: date.toDateString(),
+				repeat: isRecurring ? repeat : null,
+				repeat_unit: isRecurring ? repeatUnit : null,
+				end_date: isRecurring ? endDate.toDateString() : null,
+			})
 		close()
 	}
 
 	async function editTransaction() {
-		if (!transaction.id) {
+		if (!transaction) {
```

Comment:
```suggestion
		if (!transaction || !investment) {
```

And remove it from below